warehouse/query-core/src/main/java/datawave/query/jexl/functions/GroupingRequiredFilterFunctions.java

```diff
@@ -311,6 +313,89 @@ private static void manageMatchesInGroupRemainingArgs(Object fieldValue, String
         return Collections.unmodifiableCollection(allMatches);
     }
     
+    /**
+     * <pre>
+     * 'args' will be either a matched set of range pairs, or a matched set of range pairs followed by an index integer,
+     * Ensures that the corresponding values fit within the range , using the same paradigm as matchesinGroupLeft.
+     * </pre>
+     *
+     * @param args
+     * @return a collection of matches
+     */
+    public static Collection<?> matchesInGroupLeftRange(Object... args) {
+        if (log.isTraceEnabled()) {
+            log.trace("matchesInGroupLeftRange(" + Arrays.asList(args) + ")");
+        }
+        // positionFrom is either '0', or it is the integer value of the last argument
+        // when the argumentCount is odd
+        int positionFromLeft = 0;
+        if (args.length % 3 != 0) {
+            Object lastArgument = args[args.length - 1];
```

Comment:
much of this is boiler plate parsing for the purposes of extracting the components of the matches in group fields and bounds....